bee-common-ext/src/node.rs

```diff
@@ -11,52 +11,79 @@
 
 use crate::worker::Worker;
 
-use anymap::AnyMap;
-use futures::{channel::oneshot, future::Future};
+use anymap::{any::Any, Map};
+use futures::{
+    channel::oneshot,
+    future::{Future, FutureExt},
+};
 
 use std::{
     any::TypeId,
     collections::{HashMap, HashSet},
-    sync::Arc,
+    pin::Pin,
 };
 
-pub trait Node: Send + Sync + 'static {
+pub trait Node: Default + Send + Sync + 'static {
     fn new() -> Self;
     fn spawn<W, G, F>(&self, g: G)
     where
         Self: Sized,
         W: Worker<Self>,
         G: FnOnce(oneshot::Receiver<()>) -> F,
         F: Future<Output = ()> + Send + Sync + 'static;
+    fn worker<W>(&self) -> Option<&W>
+    where
+        Self: Sized,
+        W: Worker<Self> + Send + Sync;
+    fn set_workers(&mut self, workers: Map<dyn Any + Send + Sync>);
```

Comment:
Ngl, I wonder whether `register_resource` would be better. Shall I make this change along with my changes?

Reply:
Yeah let's do as part of your changes